### ISSUE TITLE
defaultIdle has no effect in my code

### ISSUE THREAD

**Issue by Fettabachi** [ReactTraining/react-idle#19]
I'm curious as to why Idle continues to run with `true` set. I've also tried using `false` with the same result. My intention is to use state to set the Idle variable when a function is executed so that I can disable Idle programmatically. For example - the user will be logged out and sent to the login page but I don't want Idle running on that page.
```  render() {
    const {isIdleIdle} = this.state;
    return (
      <div>
        <Idle
          defaultIdle={true}
          timeout={7000}
          onChange={({ idle}) => {
            if (idle) {
              this.showAlertOnIdle()
            }
        }}/>
      </div>
    );
  }```

**Comment by newyork-anthonyng** [comment]
@Fettabachi 

Could you elaborate on what you're asking? From what I'm reading, you're expecting the `<Idle />` component to not be present when `defaultIdle` is set to `true`?

If that is what you are trying to accomplish, you want to return something different in your render method, depending on the `idle` state.
I set up this [CodeSandbox](https://codesandbox.io/s/5mk235m41n). Let me know if this is what you're thinking of.

**Comment by Fettabachi** [comment]
here https://github.com/ReactTraining/react-idle#defaultidle it states that 

> You can start out as idle by passing `<Idle defaultIdle={true}/>`, by default it is false--assumes that loading the page initially is a user interation.

I read that as 

> if defaultIdle={true} then Idle won't run if there's no onChange event detected.

So the scenario is, if a user has been idle for 30 minutes, I pop up an alert telling them that they'll be logged out in 5 minutes if they don't perform an action. When I log them out I send them to the log in page and, if they come back to the browser tab from wherever they were, I don't want them to see an alert saying that they'll be logged out in 5 minutes since they're already logged out.

My assumption is that, after logging them out and sending them to the sign in page, if I set `<Idle defaultIdle={true}/>` on the sign in page and if it isn't being interacted with, then the pop up will not render.

**Comment by newyork-anthonyng** [comment]
@Fettabachi 

> if defaultIdle={true} then Idle won't run if there's no onChange event detected.
When you say that `Idle won't run`, do you mean that it won't `render` as well?

I updated the [CodeSandbox](https://codesandbox.io/s/5mk235m41n) example. I think it does what you're looking for.

1. When the user is inactive for 5 seconds, an alert is shown. We are using an `<Idle />` component to check for this.
2. After the alert is shown, the user has 5 seconds before they are redirected to the login page. We are using a second `<Idle />` component to check for this additional 5 seconds.
3. When the user is redirected to the login page, there is no `<Idle />` components on the page.

**Comment by Fettabachi** [comment]
No. I just don't want `this.showAlertOnIdle()` called on the log in page if there's no activity.

**Comment by AbhiSriniv** [comment]
@Fettabachi : I'm observing similar issue. ```Idle doesn't run``` when page is refreshed/reloaded with defaultIdle set to true/false.
onChange is never being called although Idle is mounted.
@newyork-anthonyng : could you please explain how you achieved **onChange is always going to get called whenever <Idle /> is on the page.** ?
Thanks for the help!

**Comment by newyork-anthonyng** [comment]
Hi @AbhiSriniv! 

Could you describe your particular use case? What is not working for you?

I just added a [new demo on Github here](https://github.com/newyork-anthonyng/react-idle-test) after I noticed the CodeSandbox example was funky. Let me know if that helps out.

**Comment by AbhiSriniv** [comment]
@newyork-anthonyng : Thanks for the reply.
I was actually assuming react-idle uses setInterval() instead of setTimeout(), I want to give warning modal to user before actually log user out. so, I was expecting handleChange to get called again (which isn't the case).
what's the best way to achieve this?

**Comment by newyork-anthonyng** [comment]
@AbhiSriniv The demo that I posted on [Github here](https://github.com/newyork-anthonyng/react-idle-test) might cover your use case. Did you take a look at it?

In the demo, the user is warned that they have been inactive after 1000ms. Then after 1500ms, they are logged out.
